afs_ioc_migration/rename.py

```diff
@@ -0,0 +1,90 @@
+import dataclasses
+import os.path
+import typing
+
+T = typing.TypeVar("T")
+ORG = "pcdshub"
+
+
+@dataclasses.dataclass(frozen=True)
+class RepoInfo:
+    """
+    Information about where an AFS repository will be transferred.
+    """
+
+    name: str
+    github_url: str
+    github_ssh: str
+    afs_source: str
+    area: str
+
+    @classmethod
+    def from_afs(cls: type[T], afs_source: str) -> T:
+        """
+        Create a RepoInfo instance given the pull path to the afs source.
+        """
+        name = rename(afs_source)
+        area = name.split("-")[1]
+        github_url = f"https://github.com/{ORG}/{name}.git"
+        github_ssh = f"git@github.com:{ORG}/{name}.git"
+        return cls(
+            name=name,
+            github_url=github_url,
+            github_ssh=github_ssh,
+            afs_source=afs_source,
+            area=area,
+        )
+
+
+afs_areas = [
+    "common",
+    "cxi",
+    "det",
+    "fee",
+    "hfx",
+    "hpl",
+    "kfe",
+    "las",
+    "lfe",
+    "mec",
+    "mfx",
+    "rix",
+    "rixs",
+    "sxr",
+    "tmo",
+    "tst",
+    "txi",
+    "ued",
+    "xcs",
+    "xpp",
+    "xrt",
+]
+
+area_renames = {
+    "rixs": "rix",
+}
+
+
+def rename(old_path: str) -> str:
+    """
+    Given an old path or partial path to an afs repository, get the new name.
+
+    The intent is to flatten filepath structures like so:
+
+    - /some/long/path/ioc/common/gigECam.git -> ioc-common-gigECam
```

Comment:
Can we save this old path somewhere?  Maybe drop it in the readme somewhere?  I think it might help people re-orient themselves with the new repo

Reply:
Yes, that's a good idea. I already have this info ready when we construct the new readme so it will be simple to drop it in.